src/zenml/cli/stack_recipes.py

```diff
@@ -0,0 +1,871 @@
+from zenml.utils.analytics_utils import AnalyticsEvent, track_event
+
+logger = get_logger(__name__)
+
+EXCLUDED_RECIPE_DIRS = [""]
+STACK_RECIPES_GITHUB_REPO = "https://github.com/zenml-io/mlops-stacks.git"
+STACK_RECIPES_REPO_DIR = "zenml_stack_recipes"
+RECIPE_RUN_SCRIPT = "run_recipe.sh"
+RECIPE_DESTROY_SCRIPT = "destroy_recipe.sh"
+SHELL_EXECUTABLE = "SHELL_EXECUTABLE"
+STACK_INFO_FILE = "stack_info.txt"
+
+
+class LocalStackRecipe:
+    """Class to encapsulate the local stack that can be run from the CLI."""
+
+    def __init__(self, path: Path, name: str) -> None:
+        """Create a new LocalStack instance.
+
+        Args:
+            name: The name of the stack, specifically the name of the folder
+                  on git
+            path: Path at which the stack is installed
+        """
+        self.name = name
+        self.path = path
+
+    @property
+    def recipes_deploy_bash_script(self) -> str:
+        """Path to the bash script that deploys the recipe.
+
+        Returns:
+            Path to the bash script that deploys the recipe.
+        """
+        return os.path.join(self.path, RECIPE_RUN_SCRIPT)
+
+    @property
+    def recipes_destroy_bash_script(self) -> str:
+        """Path to the bash script that destroys the recipe.
+
+        Returns:
+            Path to the bash script that destroys the recipe.
+        """
+        return os.path.join(self.path, RECIPE_DESTROY_SCRIPT)
+
+    # define the property here
+    @property
+    def stack_yaml_file(self) -> str:
+        """Path of the stack config YAML file created as part of recipe deployment.
+
+        Returns:
+            Path of the stack config YAML file created as part of recipe deployment.
+        """
+        stack_info_file = open(self.path / STACK_INFO_FILE, "r")
+        file_name = stack_info_file.read().replace('"', "")
+        file_name = file_name[2:]
+        stack_info_file.close()
+
+        return os.path.join(self.path, file_name.replace("\n", ""))
+
+    def is_present(self) -> bool:
+        """Checks if the stack_recipe exists at the given path.
+
+        Returns:
+            True if the stack_recipe exists at the given path, else False.
+        """
+        return fileio.isdir(str(self.path))
+
+    def run_stack_recipe(self, stack_recipe_runner: List[str]) -> None:
+        """Run the local stack recipe using the bash script at the supplied location.
+
+        Args:
+            stack_recipe_runner: Sequence of locations of executable file(s)
+                            to run the stack_recipe
+
+        Raises:
+            NotImplementedError: If the stack_recipe hasn't been implement yet.
+            FileNotFoundError: If the stack_recipe runner script is not found.
+            subprocess.CalledProcessError: If the stack_recipe runner script fails.
+        """
+        if all(map(fileio.exists, stack_recipe_runner)):
+            call = stack_recipe_runner
+            try:
+                subprocess.check_call(
```

Comment:
As suggested in the PRD I would re-think should subprocess to call a bash script. Id rather use a the `python-terraform` library and potentially do the calls here? WDYT

Reply:
I think we can switch to the library. The only thing we lose is the ability to customize commands per recipe and considering that only the `aws-minimal` recipe requires additional clean-up at the moment, I'm thinking we can just add that condition in the python code itself.